dirty_memory_manager.hh

```diff
@@ -28,12 +28,21 @@ class sstable_write_permit final {
     }
 
 public:
-    sstable_write_permit(sstable_write_permit&&) noexcept = default;
-    sstable_write_permit& operator=(sstable_write_permit&&) noexcept = default;
+    sstable_write_permit(sstable_write_permit&& o) noexcept : _permit(std::exchange(o._permit, std::nullopt)) {}
+    sstable_write_permit& operator=(sstable_write_permit&& o) noexcept {
+        if (this != &o) {
+            _permit = std::exchange(o._permit, std::nullopt);
+        }
+        return *this;
+    }
 
     static sstable_write_permit unconditional() {
         return sstable_write_permit();
     }
+
+    operator bool() const noexcept {
```

Comment:
Not motivated in the commit message. I'm not thrilled about implicit conversions operators (except for specific kinds of uses), but I'm not strongly against either.